src/mavoscript.js

```diff
@@ -533,6 +533,49 @@ var _ = Mavo.Script = {
 
 		return [a, b];
 	},
+	/**
+	 * Returns a list of all variables in the AST
+	 */
+	getVariables: function(ast) {
+		switch (ast.type) {
+			case "Literal":
+				return [];
+			case "Identifier":
+				// const isGlobal = !!globalThis[ast.name];
+				// return isGlobal? [] : [ast];	
+				return [ast];	
+			case "UnaryExpression":
+				return _.getVariables(ast.argument);
+			case "BinaryExpression":
+				if (ast.operator === ':') {
+					return _.getVariables(ast.right);
+				}
+				return [ast.left, ast.right].flatMap(_.getVariables);
+			case "LogicalExpression":
+				return [ast.left, ast.right].flatMap(_.getVariables);
+			case "CallExpression":
+				let toExplore = [...ast.arguments];
+				if (ast.callee.type !== "Identifier") {
+					toExplore = [ast.callee].concat(toExplore);
```

Comment:
Here as well, some comments about what this is trying to detect (e.g. with example expressions)